### ISSUE TITLE
No Barcode detection

### ISSUE THREAD

**Issue by Mun0n** [googlesamples/android-vision#31]
I'm using the barcode-reader project in two Samsung devices (SM-N9005 or SM-G920F) and they are not capable of recognize any barcode. There is no problems in others like One plus one, or others Samsung devices like S3 or S4. There is not a problem of the autofocus, because it is working fine, the barcode is focus properly. 

With this devices, the camera opens but nothing happens.

HELP please.

**Comment by pchx** [comment]
Does BarcodeDetector.isOperational() return true?

The devices may be low on storage space, which would prevent the actual barcode reader library from getting downloaded.

**Comment by Mun0n** [comment]
Ok, that's the problem. The device is on low storage and the barcode detector doesn't start.

But... I think if my device has 400MB of free space....is the low storage limit too low??

Zxing works with lower storage levels. What do you think?

**Comment by pchx** [comment]
Yes, we plan to tweak the free space threshold required to download the barcode detector in a future release. Stay tuned.

**Comment by Mun0n** [comment]
Thanks! Let me know as soon as you get that. This will be awesome.

Thank you guys!

**Comment by JerabekJakub** [comment]
Hi, I'm facing the same problem on Samsung S3 mini - barcodeDetector.isOperational() is false in createCameraSource() method.
I've checked free space on internal storage and I have 3,43 GB free.
I'm using latest library: com.google.android.gms:play-services:8.3.0

**Comment by omar-faruk-sust** [comment]
i am using the Nexus 5. i am facing the problem No barcode detected. is there anybody who can help to solve this problem?

**Comment by safeer-ahmed** [comment]
I am facing the same issue on ASUS Tablet running API Level 18. Is there any workaround to install them from code?

**Comment by naser9878** [comment]
devices with Android 5.1 , the camera opens but nothing happens. 
how tweak the free space threshold required to download the barcode detector ??
When will this bug be fixed?

**Comment by vegeta2102** [comment]
Hi guys!
Any update on this ?
I'm facing the same problem on Sony Xperia (OS 9)
- Vision version
'com.google.android.gms:play-services-vision:19.0.0'
- Camera were able to open but nothing happened. It seems BarcodeDetector doesn't work
